### ISSUE TITLE
Unbound: Unbound is not compiled with nghttp2

### ISSUE THREAD

**Issue by Sniper296** [openwrt/packages#22845]
Maintainer: @EricLuehrsen
Environment: ARMv7l, Linksys EA6350v3 (Civic), 21.02.2

Description:

When trying to configure Unbound to do DoT & DoH serving, I get the following error for DoH:

`daemon.warn unbound: [6099:0] warning: Unbound is not compiled with nghttp2. This is required to use DNS-over-HTTPS.`

I see nghttp2 in the Makepkg, but not in the version flags.

`unbound -V`
```
Version 1.17.1

Configure line: --target=arm-openwrt-linux --host=arm-openwrt-linux --build=x86_64-pc-linux-gnu --program-prefix= --program-suffix= --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --libexecdir=/usr/lib --sysconfdir=/etc --datadir=/usr/share --localstatedir=/var --mandir=/usr/man --infodir=/usr/info --disable-nls --disable-dsa --disable-gost --enable-allsymbols --enable-ecdsa --enable-tfo-client --enable-tfo-server --with-libexpat=/builder/shared-workdir/build/sdk/staging_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/usr --with-ssl=/builder/shared-workdir/build/sdk/staging_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/usr --with-user=unbound --with-run-dir=/var/lib/unbound --with-conf-file=/var/lib/unbound/unbound.conf --with-pidfile=/var/run/unbound.pid --enable-ipset --with-libmnl=/builder/shared-workdir/build/sdk/staging_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/usr --enable-event-api --with-libevent=/builder/shared-workdir/build/sdk/staging_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/usr --with-pthreads
Linked libs: pluggable-libevent 2.1.12-stable (it uses epoll), OpenSSL 1.1.1t  7 Feb 2023
Linked modules: dns64 ipset respip validator iterator
TCP Fastopen feature available

BSD licensed, see LICENSE in source package for details.
Report bugs to unbound-bugs@nlnetlabs.nl or https://github.com/NLnetLabs/unbound/issues
```

**Comment by brada4** [comment]
you need to build package with specific option enabled for extra functionality:

https://github.com/openwrt/packages/blob/e34369a41b522a49dec9fb1d81b4ee6b32b731d8/net/unbound/Makefile#L31

v21 is eol, but rebuild requirement stands also for current releases.

**Comment by EricLuehrsen** [comment]
I am not sure this is a wise mainstream option. DOH serving is not typical for a hobby use of OpenWrt. You need a certificate that is verifiable. DOH is not interactive so you can't agree to the warning message like LuCI unverified HTTPS. Otherwise with all the options, Unbound foot print grows and it isn't suitable replacement for dnsmasq (its already large). A professional using OpenWrt to fuel their business, and getting real certs., can easily manage customizing Makefile options in the packages.

**Comment by Sniper296** [comment]
Ah, okay. I mistook the lines in the Makefile for flags, not variables. Thanks, @brada4.

After messing around with the build system for most of the day, I eventually got a build built, and have DoH running.

My reasoning, @EricLuehrsen, is that Firefox [added ECH](https://support.mozilla.org/en-US/kb/faq-encrypted-client-hello) support recently. But requires a DoH server to use it. Until they [uncouple ECH from DoH](https://bugzilla.mozilla.org/show_bug.cgi?id=1500289), I intend to use my Unbound server. As for the TLS certificate, I used a Let's Encrypt cert for a spare domain name, so don't have to worry about the validity. I'd much rather not have to use DoH, but Mozilla aren't the fastest to get things done.

Anyway, all working now. Thanks.
